### PR TITLE
NLogEtwExtendedTarget - Added support for providing own EventSource 

### PR DESCRIPTION
**Pull request by snakefoot**
When the EventSource implements `NLogEtwExtendedTarget.INLogEventSource`

### CODE REVIEW

File: NLog.Etw/NLogEtwExtendedTarget.cs [hunk 1]
```diff
@@ -19,8 +19,33 @@ namespace NLog.Etw
     [Target("ExtendedEventTracing")]
     public sealed class NLogEtwExtendedTarget : TargetWithLayout
     {
+        /// <summary>
+        /// Interface for receiving NLog <see cref="LogEventInfo"/> for writing to custom ETW <see cref="EventSource"/>
+        /// </summary>
+        public interface INLogEventSource
```
Comment (304NotModified) [inline NLog.Etw/NLogEtwExtendedTarget.cs]: I think it should be in the outer scope? 

> X DO NOT use public nested types as a logical grouping construct; use namespaces for this.
> 
> X AVOID publicly exposed nested types. The only exception to this is if variables of the nested type need to be declared only in rare scenarios such as subclassing or other advanced customization scenarios.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/nested-types

File: NLog.Etw/NLogEtwExtendedTarget.cs [hunk 2]
```diff
@@ -53,6 +78,52 @@ public void Critical(string LoggerName, string Message)
...
+                }
+            }
+
+            EventSource INLogEventSource.EventSource { get { return this; } }
+        }
+
+        /// <summary>
+        /// Public property that allows the NLog Configuration Engine to recognize any NLog Layout on custom EventSource-implementation
+        /// </summary>
+        public INLogEventSource EventSource { get; private set; }
```
Comment (304NotModified) [inline NLog.Etw/NLogEtwExtendedTarget.cs]: can this be `public INLogEventSource EventSource { get;  }`?

File: NLog.Etw/NLogEtwExtendedTarget.cs [hunk 3]
```diff
@@ -61,44 +132,44 @@ public void Critical(string LoggerName, string Message)
         /// <param name="logEvent">event to be written.</param>
         protected override void Write(LogEventInfo logEvent)
         {
-            if (EtwLogger.Log.IsEnabled())
+            if (EventSource?.EventSource?.IsEnabled()==true)
```
Comment (304NotModified) [inline NLog.Etw/NLogEtwExtendedTarget.cs:110]: is `==true` needed? (only if IsEnabled() returns a `bool?` I think, not sure if that's the case here)

File: NLog.Etw/NLogEtwExtendedTarget.cs (hunk 1 above)
Comment (snakefoot) [reply NLog.Etw/NLogEtwExtendedTarget.cs]: Moved out. Would like to have `NLogEtwExtendedTarget` in the `NLog.Targets`-namespace, but for some reason it is not the case.

File: NLog.Etw/NLogEtwExtendedTarget.cs (hunk 2 above)
Comment (snakefoot) [reply NLog.Etw/NLogEtwExtendedTarget.cs]: Yes just noise from previous iterations. Fixed

File: NLog.Etw/NLogEtwExtendedTarget.cs (hunk 3 above)
Comment (snakefoot) [reply NLog.Etw/NLogEtwExtendedTarget.cs:110]: When using `?` then the result always become nullable, and you have to compare with "real" value.

**Comment by 304NotModified** [comment]
What's the idea behind the "code review" commit and the force pushes? I prefer OR 1 logical commit and force push, or no force push (and squash when merging).

Please note that force pushes are making reviews harder - I cannot see easily what is changed and I get notifications that give these results:

image

**Comment by snakefoot** [comment]
Me trying to rewrite history like I want it to be (Because I'm not very good at getting it right the first time). But will make it into a single commit then.

**Comment by 304NotModified** [comment]
> Me trying to rewrite history like I want it to be (Because I'm not very good at getting it right the first time). But will make it into a single commit then.

Thanks!

File: NLog.Etw/NLogEtwExtendedTarget.cs (hunk 3 above)
Comment (304NotModified) [reply NLog.Etw/NLogEtwExtendedTarget.cs:110]: thanks, missed the `?.` indeed

**Pull request merged**